scripts/auto-improve.sh

```
@@ -0,0 +1,83 @@
+# Function to wait for CodeRabbit analysis with polling
+wait_for_coderabbit() {
+    local start_time=$(date +%s)
+    local initial_commit=$(git rev-parse HEAD)
+    
+    echo "⏳ Waiting for CodeRabbit analysis (polling every ${POLL_INTERVAL}s, max ${MAX_WAIT_TIME}s)..."
+    
+    while true; do
+        local current_time=$(date +%s)
+        local elapsed=$((current_time - start_time))
+        
+        if [ $elapsed -ge $MAX_WAIT_TIME ]; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)